ShoppingM8-be/src/main/java/edu/pw/shoppingm8/receipt/Receipt.java

```diff
@@ -0,0 +1,36 @@
+package edu.pw.shoppingm8.receipt;
+
+import edu.pw.shoppingm8.list.List;
+import edu.pw.shoppingm8.user.db.User;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import javax.persistence.*;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+public class Receipt {
+    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "list_id", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
+    private List list;
```

Comment:
This field is marked as id in db changelog but not here, why?

Reply:
To avoid problems we've had with products.

If we took that approach we would again have a problem with generated id value at the moment of creation. As you can see, product changelog is made the same way - the unique constraint on id and list_id is kept by making them a composite key. Hibernate, as we've experienced, is not perfect when comes to handling this kind of situations :) So, until we have a better idea, only foreign key information is passed to hibernate, but the DB is as it should be.

Reply:
are you sure that there won't be a situation where two objects on same lists have same id field generated?

Reply:
Yes